mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -42,34 +50,122 @@ private[regression] trait LinearRegressionParams extends RegressorParams
 class LinearRegression extends Regressor[Vector, LinearRegression, LinearRegressionModel]
   with LinearRegressionParams {
 
-  setDefault(regParam -> 0.1, maxIter -> 100)
```

Comment:
minor: I think it is okay to leave the defaults here. It is easier for users to read. We put some below params because we need to have the params defined first.